website/components/LandingPage/OpenSource/index.js

```diff
@@ -55,62 +56,74 @@ function cacheSet(key, data) {
   }
 }
 
-function cachedFetch(key, url, properties) {
+function cached(key, fallback, properties) {
   const data = cacheGet(key);
   if (data) return Promise.resolve(data);
 
-  return fetch(url)
-    .then(response => response.json())
-    .then(data => {
-      if (properties) data = pick(data, properties);
-      cacheSet(key, data);
-      return data;
-    }, console.error);
+  return fallback().then(data => {
+    if (properties) data = pick(data, properties);
+    cacheSet(key, data);
+    return data;
+  }, console.error);
 }
 
-function useStats() {
-  const [stars, setStars] = useState(null);
-  const [forks, setForks] = useState(null);
-  const [downloads, setDownloads] = useState(null);
-
+function getNPMDownloads(from, to) {
   function dateRanges(from, to) {
     from = new Date(from);
     to = new Date(to);
-    const length = to.getFullYear() - from.getFullYear();
+
+    const length =
+      (to.getFullYear() - from.getFullYear()) * 12 +
+      (to.getMonth() - from.getMonth());
+
     const dates = Array.from({ length })
-      .map((_, i) => from.getFullYear() + i + '-01-01')
+      .map(() =>
+        new Date(from.setMonth(from.getMonth() + 1)).toISOString().slice(0, 10)
+      )
       .concat([to.toISOString().slice(0, 10)]);
+
     return dates
       .reduce((acc, curr, i, arr) => acc.concat(curr + ':' + arr[i + 1]), [])
       .slice(0, -1);
   }
 
+  const dates = dateRanges(from, to);
+
+  return Promise.all(
+    dates.map(range =>
+      fetch(`https://api.npmjs.org/downloads/point/${range}/uniforms`)
+        .then(response => response.json())
+        .then(({ downloads }) => downloads || 0)
+    )
+  ).then(sums => sums.reduce((acc, curr) => acc + curr));
```

Comment:
This fires quite a lot of requests at once. We can keep it as it is now, but let's just keep it in mind.